HeterogeneousCore/CUDAUtilities/interface/currentDevice.h

```diff
@@ -0,0 +1,16 @@
+#ifndef HeterogenousCore_CUDAUtilities_currentDevice_h
+#define HeterogenousCore_CUDAUtilities_currentDevice_h
+
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+
+#include <cuda_runtime.h>
+
+namespace cudautils {
+  inline int currentDevice() {
```

Comment:
@waredjeb once the PR is merged, you may reuse this instead of defining an ad hoc function